diff --git a/config/android/BUILD.gn b/config/android/BUILD.gn
index 8c77eb2c9..a6f72f1ba 100644
--- a/config/android/BUILD.gn
+++ b/config/android/BUILD.gn
@@ -92,6 +92,7 @@ config("compiler") {
 # that is Android-only. Please see that target for advice on what should go in
 # :runtime_library vs. :compiler.
 config("runtime_library") {
+  
 }
 
 config("hide_all_but_jni_onload") {
diff --git a/config/compiler/BUILD.gn b/config/compiler/BUILD.gn
index 33bda0935..25a870aa8 100644
--- a/config/compiler/BUILD.gn
+++ b/config/compiler/BUILD.gn
@@ -1222,6 +1222,7 @@ config("compiler_cpu_abi") {
         ]
       }
     } else if (current_cpu == "arm") {
+      ldflags += [ "-Wl,--hash-style=sysv" ]
       if (is_clang && !is_android && !is_nacl && !is_chromeos_device) {
         cflags += [ "--target=arm-linux-gnueabihf" ]
         ldflags += [ "--target=arm-linux-gnueabihf" ]
@@ -1236,6 +1237,7 @@ config("compiler_cpu_abi") {
         cflags += [ "-mtune=$arm_tune" ]
       }
     } else if (current_cpu == "arm64") {
+      ldflags += [ "-Wl,--hash-style=sysv" ]
       if (is_clang && !is_android && !is_nacl && !is_fuchsia &&
           !is_chromeos_device) {
         cflags += [ "--target=aarch64-linux-gnu" ]
