diff --git a/BUILD.gn b/BUILD.gn
index 1e3f0181dfb..9d1583f5ac8 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -1154,6 +1154,16 @@ config("cppgc_header_features") {
   }
 }
 
+config("android15_16kb_support") {
+  ldflags = []
+  if (v8_current_cpu == "x64" || v8_current_cpu == "arm64") {
+    ldflags += [
+      "-Wl,-z,max-page-size=16384",
+      "-Wl,-z,common-page-size=16384",
+    ]
+  }
+}
+
 enabled_external_defines =
     enabled_external_v8_defines + enabled_external_cppgc_defines
 disabled_external_defines =
@@ -6477,7 +6487,7 @@ v8_source_set("torque_ls_base") {
   ]
 }
 
-v8_component("v8_libbase") {
+v8_source_set("v8_libbase") {
   sources = [
     "src/base/abort-mode.cc",
     "src/base/abort-mode.h",
@@ -6809,7 +6819,7 @@ if (v8_use_libm_trig_functions) {
   }
 }
 
-v8_component("v8_libplatform") {
+v8_source_set("v8_libplatform") {
   sources = [
     "include/libplatform/libplatform-export.h",
     "include/libplatform/libplatform.h",
diff --git a/src/base/build_config.h b/src/base/build_config.h
index 207f46b057d..48436737083 100644
--- a/src/base/build_config.h
+++ b/src/base/build_config.h
@@ -96,8 +96,8 @@ constexpr int kMinimumOSPageSize = 16 * 1024;
 // pages.
 constexpr int kMinimumOSPageSize = 64 * 1024;
 #else
-// Everything else uses 4kB pages.
-constexpr int kMinimumOSPageSize = 4 * 1024;
+// Everything else uses 16kB pages.
+constexpr int kMinimumOSPageSize = 16 * 1024;
 #endif
 
 #endif  // V8_BASE_BUILD_CONFIG_H_
